File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        first, last = 0, len(numbers) - 1

        while first < last:
            current_sum = numbers[first] + numbers[last]

            if current_sum == target:
                return [first + 1, last + 1]

            if current_sum > target:
                last -= 1

            else:
                first += 1

        return [-1, -1]

File: Data Structures & Algorithms/strobogrammatic-number/submission-0.java
class Solution {
    public boolean isStrobogrammatic(String num) {
        Map<Character, Character> hash = new HashMap<>();

        hash.put('0', '0');
        hash.put('8', '8');
        hash.put('1', '1');
        hash.put('6', '9');
        hash.put('9', '6');

        int first = 0, last = num.length() - 1;

        while (first <= last) {
            if (num.charAt(first) != hash.getOrDefault(num.charAt(last), 'Z')) return false;
            first++;
            last--;
        }

        return true;
    }
}
